pycaret/internal/pycaret_experiment/supervised_experiment.py

```diff
@@ -5541,3 +5555,173 @@ def deep_check(self, estimator, check_kwargs: Optional[dict]):
+
+
+        path_or_file: str or BinaryIO (file pointer)
+            The path/file pointer to load the experiment from.
+            The pickle file must be created through ``save_experiment``.
+
+
+        data: dataframe-like
+            Data set with shape (n_samples, n_features), where n_samples is the
+            number of samples and n_features is the number of features. If data
+            is not a pandas dataframe, it's converted to one using default column
+            names.
+
+
+        data_func: Callable[[], DATAFRAME_LIKE] = None
+            The function that generate ``data`` (the dataframe-like input). This
+            is useful when the dataset is large, and you need parallel operations
+            such as ``compare_models``. It can avoid broadcasting large dataset
+            from driver to workers. Notice one and only one of ``data`` and
+            ``data_func`` must be set.
+
+
+        test_data: dataframe-like or None, default = None
+            If not None, test_data is used as a hold-out set and `train_size` parameter
+            is ignored. The columns of data and test_data must match.
+
+
+        preprocess_data: bool, default = True
+            If True, the data will be preprocessed again (through running ``setup``
+            internally). If False, the data will not be preprocessed. This means
+            you can save the value of the ``data`` attribute of an experiment
+            separately, and then load it separately and pass it here with
+            ``preprocess_data`` set to False. This is an advanced feature.
+            We recommend leaving it set to True and passing the same data
+            as passed to the initial ``setup`` call.
+
+
+        **cloudpickle_kwargs:
+            Kwargs to pass to the ``cloudpickle.load`` call.
+
+
+        Returns:
+            loaded experiment
+
+        """
+
+        return cls._load_experiment(
+            path_or_file,
+            cloudpickle_kwargs=cloudpickle_kwargs,
+            preprocess_data=preprocess_data,
+            data=data,
+            data_func=data_func,
+            test_data=test_data,
+        )
+
+    @property
+    def X(self):
+        """Feature set."""
+        return self.dataset.drop(self.target_param, axis=1)
+
+    @property
+    def dataset_transformed(self):
+        """Transformed dataset."""
+        return pd.concat([self.train_transformed, self.test_transformed])
+
+    @property
+    def X_train_transformed(self):
+        """Transformed feature set of the training set."""
+        return self.pipeline.transform(
+            X=self.X_train,
+            y=self.y_train,
+            filter_train_only=False,
+        )[0]
+
+    @property
+    def train_transformed(self):
+        """Transformed training set."""
+        return pd.concat(
+            [self.X_train_transformed, self.y_train_transformed],
+            axis=1,
+        )
+
+    @property
+    def X_transformed(self):
```

Comment:
This should be abstract property since the implementation is different between Regression/Classification and Time Series.